stestr/commands/run.py

```diff
@@ -308,6 +308,11 @@ def run_command(config='.stestr.conf', repo_type='file',
             exit(1)
         repo = util.get_repo_initialise(repo_type, repo_url)
     combine_id = None
+    if concurrency and concurrency < 0:
+        msg = ("The provided concurrency value: %s is not valid. An integer "
+               ">= 0 must be used.\n")
+        stdout.write(msg)
+        return 2
```

Comment:
And, if we run stestr with `--slowest` option, slowest test info are showed in this case. So, should we check this return value there?
https://github.com/mtreinish/stestr/pull/208/files#diff-e77977ac6a427e77fc880ed584215c86R197

Reply:
Hmm, this is an interesting case. Since the return code is non-zero for expected error conditions (like a test failure) where we probably still want to print the slowest tests (when `--slowest` is set). 

I agree that we don't want to print the slowest in the case like this with an invalid input and fails before we run any tests, but I'm not sure of a way to perfectly detect that. Were you thinking check for the unique return code `2` here? I don't think all of these edge conditions will have unique return codes to detect.

Reply:
well, yeah. I see. So maybe, it's better to check it in the earlier, like after store the concurrency value?
https://github.com/mtreinish/stestr/pull/208/files#diff-e77977ac6a427e77fc880ed584215c86R172

Reply:
Ah, yeah that's a good call I'll copy the check there too. We need to keep this inside the method too for users using the python API instead of the CLI. But, we'll never reach the inner call via the cli if the input is bad so it'll avoid the issue you're outlining here.